cmd/gazelle/BUILD.bazel

```diff
@@ -34,14 +34,10 @@ go_library(
 go_binary(
     name = "gazelle",
     embed = [":go_default_library"],
-    visibility = ["//visibility:public"],
-)
-
-go_binary(
-    name = "gazelle_pure",
-    embed = [":go_default_library"],
+    msan = "off",
     pure = "on",
-    tags = ["manual"],
+    race = "off",
+    static = "off",
```

Comment:
I'm okay with disabling `msan`, `race`, and `static` in `//cmd/gazelle:gazelle` (though I'm not sure I understand the motivation), but I'm not okay with `pure = "on"` in `//cmd/gazelle:gazelle`. Keep `gazelle_pure` and use that if you need it.

Pure mode forces us to compile the standard library in a different mode so that it doesn't include any cgo code. That can slow down builds considerably if both pure and non-pure targets are being built.

Reply:
The motivation is to avoid the dependency on `--feature` flags for gazelle to avoid re-building it if a flag is used. Also (for pure) to avoid depending on libc (fixes rule usage on nixos).
I agree with the point re. building stdlib in pure mode - I've reverted the change. I guess we have to use non-rule gazelle on nix.
I've explicitly set pure=off to avoid depending to the `--feature` flag.

Reply:
Actually, I've just found out that we could do the following to run a pure version of gazelle via the rule:
```bzl
gazelle(
    name = "gazelle",
    gazelle = "@bazel_gazelle//cmd/gazelle:gazelle_pure",
)
```

Reply:
Ah, ok, avoiding a dependency on feature flags makes more sense now.

+1 to using the `gazelle` attribute in the `gazelle` rule. Seems like the best way to do this.